src/docs/components/markdown/pre.svelte

```diff
@@ -0,0 +1,48 @@
+<script lang="ts">
+	import Check from '~icons/lucide/check';
+	import Copy from '~icons/lucide/copy';
+	import { fly } from 'svelte/transition';
+	import { cn } from '$docs/utils';
+
+	let codeString: string;
+
+	let className: string | undefined | null = undefined;
+	export { className as class };
+
+	let copied = false;
+	let copytimeout: ReturnType<typeof setTimeout>;
+
+	function copyCode() {
+		navigator.clipboard.writeText(codeString);
+		copied = true;
+		clearTimeout(copytimeout);
+		copytimeout = setTimeout(() => {
+			copied = false;
+		}, 2500);
+	}
+
+	function copyCodeToClipboard(node: HTMLPreElement) {
+		codeString = node.innerText.trim() ?? '';
+	}
```

Comment:
We use this in some places. Guess it'd make sense to abstract it maybe, same with the copy icon below.

> Copy builder incoming? 👀

Reply:
Yeah I don't think that's a bad idea actually! Could be more of a "utility"!

Reply:
Added an initial internal 'copy button builder' to the doc utils, seems like something we could def consider opening to the public as well!